tests/testthat/test-financial_functions.R

```diff
@@ -1,12 +1,28 @@
 test_that("calculate_net_profits penalizes companies for late build out of low
           carbon technologies", {
   input_data <- tibble::tribble(
-    ~company_name, ~baseline, ~late_sudden, ~Baseline_price, ~late_sudden_price, ~net_profit_margin, ~direction, ~proximity_to_target,
-    "leader", 100, 150, 10, 10, 0.1, "increasing", 1,
-    "laggard", 100, 150, 10, 10, 0.1, "increasing", 0.5
+    ~company_name, ~baseline, ~late_sudden, ~Baseline_price, ~late_sudden_price, ~net_profit_margin, ~direction, ~proximity_to_target, ~scenario_geography, ~year, ~emission_factor,
```

Comment:
So far you only adapted the test so they still run but did not add a new one to test specifically the behavior of the carbon tax.
How about you add a unit test where you calculate net_profits on the same input data with or without carbon tax and make some assuptions on the differences (happy to explain in more detail)

Reply:
that would be amazing!